optimum/onnxruntime/modeling_diffusion.py

```diff
@@ -538,181 +612,239 @@ def forward(
         text_embeds: Optional[Union[np.ndarray, torch.Tensor]] = None,
         time_ids: Optional[Union[np.ndarray, torch.Tensor]] = None,
         timestep_cond: Optional[Union[np.ndarray, torch.Tensor]] = None,
+        cross_attention_kwargs: Optional[Dict[str, Any]] = None,
+        added_cond_kwargs: Optional[Dict[str, Any]] = None,
+        return_dict: bool = False,
     ):
         use_torch = isinstance(sample, torch.Tensor)
 
+        if len(timestep.shape) == 0:
+            timestep = timestep.unsqueeze(0)
+
         model_inputs = {
             "sample": sample,
             "timestep": timestep,
             "encoder_hidden_states": encoder_hidden_states,
             "text_embeds": text_embeds,
             "time_ids": time_ids,
             "timestep_cond": timestep_cond,
+            **(cross_attention_kwargs or {}),
+            **(added_cond_kwargs or {}),
         }
 
-        onnx_inputs = self._prepare_onnx_inputs(use_torch, **model_inputs)
+        onnx_inputs = self.prepare_onnx_inputs(use_torch, **model_inputs)
         onnx_outputs = self.session.run(None, onnx_inputs)
-        model_outputs = self._prepare_onnx_outputs(use_torch, *onnx_outputs)
+        model_outputs = self.prepare_onnx_outputs(use_torch, *onnx_outputs)
+
+        if return_dict:
+            return model_outputs
+
+        return ModelOutput(**model_outputs)
+
+    @property
+    def add_embedding(self):
```

Comment:
why not just override `_get_add_time_ids` and skip that dimension check part, feel it could be a bit hard-coded here.

Reply:
yes that's what I did at first, then I thought this is more generalizable, but yeh it is too hacky, reverting it.